common.sh

```diff
@@ -49,9 +49,26 @@ NODES_PLATFORM=${NODES_PLATFORM:-"libvirt"}
 MASTER_NODES_FILE=${MASTER_NODES_FILE:-"ocp/master_nodes.json"}
 
 export RHCOS_IMAGE_URL=${RHCOS_IMAGE_URL:-"https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/"}
+
+let _builds_count=$(echo $(curl ${RHCOS_IMAGE_URL}/builds.json | jq -r ".builds|length") - 2 | bc)
+
 export RHCOS_LATEST="$(curl ${RHCOS_IMAGE_URL}/builds.json | jq -r ".builds[0]")"
 export RHCOS_IMAGE_VERSION="${RHCOS_IMAGE_VERSION:-${RHCOS_LATEST}}"
 export RHCOS_IMAGE_FILENAME_OPENSTACK_GZ="$(curl ${RHCOS_IMAGE_URL}/${RHCOS_IMAGE_VERSION}/meta.json | jq -r '.images.openstack.path')"
+
+if [[ -z $RHCOS_IMAGE_FILENAME_OPENSTACK_GZ || "$RHCOS_IMAGE_FILENAME_OPENSTACK_GZ" == "null" ]]; then
+    for _i in $( seq -s ' ' 1 ${_builds_count}); do
+        export RHCOS_LATEST="$(curl ${RHCOS_IMAGE_URL}/builds.json | jq -r ".builds[$_i]")"
+        export RHCOS_IMAGE_VERSION="${RHCOS_LATEST}"
+        export RHCOS_IMAGE_FILENAME_OPENSTACK_GZ="$(curl ${RHCOS_IMAGE_URL}/${RHCOS_IMAGE_VERSION}/meta.json | jq -r '.images.openstack.path')"
+        if [[ -z $RHCOS_IMAGE_FILENAME_OPENSTACK_GZ || "$RHCOS_IMAGE_FILENAME_OPENSTACK_GZ" == "null" ]]; then
```

Comment:
If we expect this to be a temporary problem, would it be simpler to just pin to a specifc known-good build pending resolution of the missing images?

cc @karmab and @gklein

Reply:
i think it s better to use a pinned version and if latest is specified and not working, to actually report it

Reply:
Given @hardys suggestion, maybe it would be better to save this extra logic and just move to a safer location at:
https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/

See my comment here:
https://github.com/openshift-metalkube/dev-scripts/issues/347#issuecomment-483204813

Reply:
That's a temporary hack I used to test local things, that's why I marked it as _WIP_ with a _DNM_ message :smile: I'm OK with pinning the image or image building fix. Anyway, thanks for reviews :+1:

Reply:
I get AccessDenied trying to reach https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/ - what sort of authentication do we require to switch to that?

Reply:
> I get AccessDenied trying to reach https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/ - what sort of authentication do we require to switch to that?
I believe there is no index page accessible, but I had no problem accessing any of the images directly.

See my comment here:
https://github.com/openshift-metalkube/dev-scripts/issues/347#issuecomment-483227137

Reply:
nvm I see the script in #347 now, the issue seems to be there's no UX for browsing the version anymore, so we've got to use the json file.

Reply:
> nvm I see the script in #347 now, the issue seems to be there's no UX for browsing the version anymore, so we've got to use the json file.
Right, I guess our best "index" page is:
https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/builds.json